src/main/java/nl/knaw/dans/ttv/core/Task.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.ttv.core;
+
+import nl.knaw.dans.ttv.db.TransferItemDAO;
+
+import java.util.concurrent.Callable;
+
+public abstract class Task implements Callable<TransferItem> {
```

Comment:
Nice find, this `Callable` interface! That seems indeed to be exactly what it needed here.